### PR TITLE
Fix many warnings from errorprone

### LINKED ISSUE #3061: Many warnings from errorprone
Clearing away a long list of deprecation warnings exposes a number of warnings and suggestions from `errorprone`.  Addressing these would not only improve code quality, but make it easier to see the remaining compiler warnings.

### PR DESCRIPTION
**Pull request by mwoodiupui**
## References
* Fixes #3061

## Description
Many small fixes for warnings and suggestions by `errorprone`.  Also the usual IDE suggestions, tidying commentary, etc.

## Instructions for Reviewers
The aim of this patch is to reduce the long scroll of warnings from a successful build to manageable size, and make the remaining ones easier to notice so that they get some attention.  What you should see is that the output of a build is not so overwhelming.  No new features or specific bug fixes are included, though some of the fixed code was probably buggy.  Some of the affected code should be easier to understand.  There may be slight performance improvements.

This patch includes most of the commits from #2956 because without them many of the `errorprone` warnings would not be shown.  (There is a limit in the Java compiler on the numbers of errors and warnings shown.  This suggests that **we have not been seeing all of the warnings**.  We should **keep** these trimmed down so that we don't miss something that's worse than annoying.)

## Checklist
- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR modifies the REST API, I've linked to the REST Contract page (or open PR) related to this change.

### CODE REVIEW

**Comment by tdonohue** [comment]
@mwoodiupui : Are you still planning on working on this soon?  If so, I can leave this as tentative for the 7.0 release.  Otherwise, I'll likely move this to 7.1 (as we'll be working to try to get 7.0 out as quickly as reasonable after Testathon).

**Comment by mwoodiupui** [comment]
Let's move it to 7.1.

**Comment by hardyoyo** [comment]
I'm interested in helping out with this, in whatever way I can, please let me know if I can be of assistance.

**Comment by mwoodiupui** [comment]
@hardyoyo and others:  this is about all that I feel confident to fix.  There are scads of StringSplitter warnings, because I'm unsure whether the code is depending on `split`'s "surprising behavior".  I'm not sure about the HidingField or TypeParameterUnusedInFormals warnings either.  There are a handful of others.  I'm hoping that someone who really knows these classes will address the remaining warnings, now that they are not drowning in a river of more obvious stuff.

**Review by tdonohue — COMMENTED** [review]
@mwoodiupui : Did a code review today & the code looks good.  A few very minor inline comments below (no show stoppers). I have not had a chance to test it yet, which I'd like to do  before final approval & merging.

It also looks to me like this fixes #7935, so I've added that to the description of this PR.

File: dspace-api/src/main/java/org/dspace/app/bulkedit/DSpaceCSV.java [hunk 1]
```diff
@@ -577,7 +577,7 @@ public final void addItem(String line) throws Exception {
                 csvLine.add(headings.get(i - 1), null);
                 String[] elements = part.split(escapedValueSeparator);
                 for (String element : elements) {
-                    if ((element != null) && (!"".equals(element))) {
+                    if ((element != null) && !"".equals(element)) {
```
[the remark this answers is not in the document]
Comment (mwoodiupui) [reply dspace-api/src/main/java/org/dspace/app/bulkedit/DSpaceCSV.java:580]: Yes, well caught.

File: dspace-api/src/main/java/org/dspace/core/I18nUtil.java [hunk 2]
```diff
@@ -316,18 +314,17 @@ private static String getFilename(Locale locale, String fileName, String fileTyp
...
             fileType = "";
         }
 
-        if (!("".equals(locale.getCountry()))) {
+        if (!"".equals(locale.getCountry())) {
             fileNameLC = fileName + "_" + locale.getLanguage() + "_"
                 + locale.getCountry();
 
-            if (!("".equals(locale.getVariant()))) {
+            if (!"".equals(locale.getVariant())) {
```
[the remark this answers is not in the document]
Comment (mwoodiupui) [reply dspace-api/src/main/java/org/dspace/core/I18nUtil.java:327]: Yes.

**Comment by mwoodiupui** [comment]
Hm, I did not intend to include `log4j2-test.xml` in this patch.  I want to do some testing, and remove the old, now unused `log4j.properties`.  I've removed the new file from this PR and unlinked the related issue.

**Review by tdonohue — APPROVED** [review]
👍 Looks good now @mwoodiupui !  Thank you again for all this general code cleanup.  The changes all make sense to me, and I also built/deployed successfully with no issues.

**Pull request merged**